test/src/main/java/org/apache/accumulo/test/ScanServerConcurrentTabletScanIT.java

```diff
@@ -135,7 +157,15 @@ public void testScanSameTabletDifferentDataTabletMetadataCacheEnabled() throws E
       // Ingest another 100 k/v with a different column family
       final int secondBatchOfEntriesCount = ingest(client, tableName, 10, 10, 0, "COLB", true);
 
+      // Add a sleep that is long enough that the configured refresh interval passes if
+      // the test has been set to use one.
+      Thread.sleep(1000);
```

Comment:
fwiw, if you can mock time to avoid flaky tests then Caffeine's time source is configurable ([docs](https://github.com/ben-manes/caffeine/wiki/Testing)). It may not be practical if the code wasn't designed for that, of course. If it is then Guava's `FakeTicker` is very handy, as is using a caller-runs executor (`Runnable::run`) or `Awaitility` for coordinating concurrent calls.

Reply:
Thanks for the tip for using the mock time, I don't think it would be easy to set in this case but would be useful for other cases for sure. We also have a [Wait.waitFor() ](https://github.com/apache/accumulo/blob/861d28b64ba80b82300306df148ea3f969e99e6d/test/src/main/java/org/apache/accumulo/test/util/Wait.java#L87) utility which could be used here instead of sleep by scanning in a loop probably, at least to check for the refresh case (wouldn't help so much for checking it didn't refresh)

Reply:
oh neat. What are the benefits of that utility versus [Awaitility](http://www.awaitility.org/)?

Reply:
`Awaitility` looks to be a lot better but I didn't even know about it until you pointed it out. The `Wait` utility class is just a simple utility to wait for a condition to turn true that we added for help with testing but it appears like we could replace it with `Awaitility` and get a lot more features.